fbw-a380x/src/systems/systems-host/index.ts

```diff
@@ -154,8 +195,21 @@ class SystemsHost extends BaseInstrument {
 
         this.soundManager.update(dt);
         this.gpws.update(dt);
-        this.fwsCore.update(dt);
+        this.fwsCore?.update(dt);
       });
+
+    this.fwsAvailable.sub((a) => {
+      if (!a && this.fwsCore !== undefined) {
+        this.fwsCore = undefined;
+      } else if (a && this.fwsCore === undefined) {
+        this.fwsCore = new FwsCore(1, this.bus, this.failuresConsumer, this.fws1Failed, this.fws2Failed);
+        this.fwsCore.init();
+      }
+    }, true);
+    this.fws1Failed.sub((f) => SimVar.SetSimVarValue('L:A32NX_FWS1_IS_HEALTHY', SimVarValueType.Bool, !f));
```

Comment:
Ideally these would be set inside the FWS itself after the startup completed/shutdown but since the code keeps running regardless of that timer I guess we can ignore it.